File: src/Domain/Entities/TestRun.cs
using Domain.Enums;

namespace Domain.Entities;

public class TestRun : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string Environment { get; set; } = string.Empty;
    public TestRunStatus Status { get; set; } = TestRunStatus.Active;
    public Guid? ExecutedById { get; set; }

    public Guid ProjectId { get; set; }
    public Project Project { get; set; } = null!;

    public ICollection<TestResult> TestResults { get; set; } = [];
}

File: apps/api/src/domain/test-run.ts
import { TestRunStatus } from "@testcraft/types";

export interface TestRun {
  id: string;
  projectId: string;
  name: string;
  environment: string;
  status: TestRunStatus;
  source: string | null;
  executedById: string | null;
  createdAt: Date;
  updatedAt: Date;
}
